lib/datasource/golang-version/index.ts

```diff
@@ -0,0 +1,114 @@
+
+  @cache({ namespace: `datasource-${GolangVersionDatasource.id}`, key: 'all' })
+  async getReleases({
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    const res: ReleaseResult = {
+      homepage: 'https://go.dev/',
+      sourceUrl:
+        'https://github.com/golang/website/blob/master/internal/history/release.go',
+      releases: [],
+    };
+    const golangVersionsUrl = `${registryUrl}master/internal/history/release.go`;
+
+    try {
+      const response = await this.http.get(golangVersionsUrl);
+
+      const lines = response.body.split(lineTerminationRegex);
+
+      const startOfReleases = lines.indexOf('var Releases = []*Release{');
+      if (startOfReleases === -1) {
+        throw new Error('Invalid file - could not find the Releases section');
+      }
+
+      // Remove part before releases
+      lines.splice(0, startOfReleases + 1);
+
+      // Parse the release list
+      let release: Release = { version: undefined };
+      while (lines.length !== 0) {
+        const line = lines.shift().trim();
+
+        if (line === releaseBeginningChar) {
+          // istanbul ignore if
+          if (release.version !== undefined) {
+            throw new Error(
+              'Invalid file - unexpected error while parsing a release'
+            );
+          }
+        } else if (line === releaseTerminationChar) {
+          // istanbul ignore if
+          if (release.version === undefined) {
+            throw new Error('Invalid file - release has empty version');
+          }
+          res.releases.push(release);
+          release = { version: undefined };
+        } else {
+          const releaseDateMatch = releaseDateRegex.exec(line);
+          if (releaseDateMatch) {
+            // Make a valid UTC timestamp
+            const year = releaseDateMatch.groups.year.padStart(4, '0');
+            const month = releaseDateMatch.groups.month.padStart(2, '0');
+            const day = releaseDateMatch.groups.day.padStart(2, '0');
+            release.releaseTimestamp = `${year}-${month}-${day}T00:00:00.000Z`;
+          }
+          const releaseVersionMatch = releaseVersionRegex.exec(line);
+          if (releaseVersionMatch) {
+            release.version = `${releaseVersionMatch.groups.versionMajor}.${releaseVersionMatch.groups.versionMinor}.${releaseVersionMatch.groups.patch}`;
+            // istanbul ignore if
+            if (!isVersion(release.version)) {
+              throw new Error(
+                `Version ${release.version} is not a valid semver`
+              );
+            }
+          }
+        }
+      }
+
+      // istanbul ignore if
+      if (res.releases.length === 0) {
+        throw new Error(`Invalid file - zero releases extracted`);
+      }
+    } catch (err) {
+      this.handleGenericErrors(err);
+    }
+
+    return res;
+  }
+
+  override handleSpecificErrors(err: HttpError): never | void {
+    throw new ExternalHostError(err);
+  }
```

Comment:
I think we don't need this.
```suggestion
```

Reply:
@viceice without this block the datasource returns `null` in case of empty results and error 404. Is that acceptable?

Reply:
Please ignore my previous message, it is incorrect.